uni/lib/model/entities/calendar_event.dart

```diff
@@ -1,12 +1,35 @@
+import 'package:intl/intl.dart';
+
 /// An event in the school calendar
 class CalendarEvent {
   /// Creates an instance of the class [CalendarEvent]
-  CalendarEvent(this.name, this.date);
+  CalendarEvent(this.name, this.date, {this.closeDate}) {
+    name = name;
+    date = date;
+    closeDate = getDateTime();
+  }
   String name;
   String date;
+  DateTime? closeDate;
```

Comment:
What is this `closeDate`?

Reply:
It is a non exact date conversion of the string, as not all strings are formatted in the same way (e.g "02 de fevereiro", "Até 05 de maio", "De 02 a 05 de março")

Reply:
Hm, I would convert it to a getter, perhaps `get parsedStartDate`, and calculated when needed.